asv/commands/publish.py

```diff
@@ -128,6 +129,28 @@ def run(cls, conf):
                 machines[d['machine']] = d
 
         log.step()
+        log.info("Getting tags and branches")
+        with log.indent():
+            repo = get_repo(conf)
+            repo.pull()
+            tags = {}
+            for tag in repo.get_tags():
+                log.dot()
+                tags[tag] = repo.get_date_from_name(tag)
+
+            branch_cache = BranchCache(conf, repo)
+
+            def iter_branches(results):
```

Comment:
It's a bit confusing to have this function defined here but used a number of lines below.